firedrake/interpolation.py

```diff
@@ -215,8 +250,58 @@ def __init__(
         self._allow_missing_dofs = allow_missing_dofs
         self.callable = None
 
+    @PETSc.Log.EventDecorator()
+    def interpolate(self, *function, output=None, transpose=False, default_missing_val=None):
+        """This performs interpolation by assembling `Interp`, which in turn calls
+        `Interpolator._interpolate`. Having this structure ensures consistency between
+        `Interpolator` and `Interp`.
+
+        This mechanism handles annotation since performing interpolation will drop an
+        Assemble block on the tape.
+
+        :arg function: If the expression being interpolated contains an
+            :class:`ufl.Argument`, then the :class:`.Function` value to
+            interpolate.
+        :kwarg output: Optional. A :class:`.Function` to contain the output.
+        :kwarg transpose: Set to true to apply the transpose (adjoint) of the
+              interpolation operator.
+        :returns: The resulting interpolated :class:`.Function`.
```

Comment:
Note that you are missing documentation on `default_missing_val`. As in the other comment, you can rectify this by just reusing the existing docstring